addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -276,3 +276,62 @@ def test_pol_description(self):
 
         po = self.env['purchase.order'].search([('partner_id', '=', self.partner_vendor_service.id)], order='id desc', limit=1)
         self.assertEqual(po.order_line.name, "[C01] Name01")
+
+    def test_pol_custom_attribute(self):
+        """
+         test that custom atributes are passed from the SO the PO for service products
+        """
+        # Setup service product variants
+        service = self.env['product.product'].create({
+            'name': 'Super Product',
+            'type': 'service',
+            'service_to_purchase': True,
+            'seller_ids': [(0, 0, {
+                'name': self.partner_vendor_service.id,
+                'min_qty': 1,
+                'price': 10,
+                'sequence': 1,
+            })]
+        })
```

Comment:
Make sure of using already created records in the tests whenever possible, as creating a new set of records for each test would drastically slow down the whole process.
Here you could make use of `self.service_purchase_1` instead.